src/ui/public/embeddable/embeddable_factory.ts

```diff
@@ -26,6 +26,17 @@ export type OnEmbeddableStateChanged = (embeddableStateChanges: EmbeddableState)
  * The EmbeddableFactory creates and initializes an embeddable instance
  */
 export abstract class EmbeddableFactory {
+  public name: string;
+
+  /**
+   *
+   * @param name - a unique identified for this factory, which will be used to map an embeddable spec to
+   * a factory that can generate an instance of it.
+   */
+  constructor(name: string) {
```

Comment:
**Question**: Do we anticipate this interface changing again in the future? If so, would it be worth using `constructor({ name })` vs `constructor(name)`?

Reply:
yeaa, lets do it, good idea.